taxi/tests/test_views.py

```diff
@@ -0,0 +1,218 @@
+            username="test",
+            password="12345",
+            license_number="TTT12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="1_test_manufacturer")
+        Manufacturer.objects.create(name="2_test_manufacturer")
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(res.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(res.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(res, "taxi/manufacturer_list.html")
+
+    def test_create_manufacturer(self):
+        data = {"name": "test manufacturer", "country": "Test country"}
+        res = self.client.post(reverse("taxi:manufacturer-create"), data=data)
+        manufacturer = Manufacturer.objects.get(name="test manufacturer")
+        self.assertEqual(manufacturer.name, data["name"])
+        self.assertEqual(manufacturer.country, data["country"])
+        self.assertRedirects(res, reverse("taxi:manufacturer-list"))
+
+    def test_update_manufacturer(self):
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        update_data = {"name": "new test", "country": "test country"}
+        url = reverse("taxi:manufacturer-update", args=[manufacturer.id])
+        res = self.client.post(url, data=update_data)
+        manufacturer.refresh_from_db()
+        self.assertEqual(manufacturer.name, update_data["name"])
+        self.assertRedirects(res, reverse("taxi:manufacturer-list"))
+
+    def test_delete_manufacturer(self):
+        manufacturer = Manufacturer.objects.create(
+            name="test name",
+            country="test country"
+        )
+        url = reverse("taxi:manufacturer-delete", args=[manufacturer.id])
+        res = self.client.post(url)
+        self.assertRedirects(res, reverse("taxi:manufacturer-list"))
+
+        with self.assertRaises(Manufacturer.DoesNotExist):
+            Manufacturer.objects.get(id=manufacturer.id)
+
+    def test_search_manufacturer(self):
+        Manufacturer.objects.create(name="1_test_manufacturer")
+        Manufacturer.objects.create(name="2_test_manufacturer")
+        res = self.client.get(
+            MANUFACTURER_URL, {"name": "1_test_manufacturer"}
+        )
+        self.assertEqual(res.status_code, 200)
+        manufacturers = Manufacturer.objects.filter(
+            name__icontains="1_test_manufacturer"
+        )
+        self.assertEqual(
+            list(res.context["manufacturer_list"]),
+            list(manufacturers),
+        )
+
+
+class PublicCarTest(TestCase):
+    def test_loggin_required_car_list(self):
+        res = self.client.get(CAR_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+    def test_loggin_required_car_detail(self):

```

Comment:
There's a typo in the method name `test_loggin_required_car_detail`. It should be `test_login_required_car_detail` for clarity and correctness.